packages/common/src/hardhat/plugins/ExtendedWeb3.js

```diff
@@ -0,0 +1,84 @@
+// Pluging notes:
+// The plugin adds 4 methods to the hre: getContract, findEvent, assertEventEmitted, assertEventNotEmitted.
+// It requires the web3 plugin to work. All contract instantiations are web3, not truffle.
+//
+// getContract usage:
+//   const { getContract } = hre;
+//   const Voting = getContract("Voting");
+//
+//   // Get deployed voting contract for network.
+//   const voting = await Voting.deployed();
+//
+//   // Create new Voting contract.
+//   const voting = await Voting.new(arg1, arg2, ...).send({ from: accounts[0] });
+//
+//   // Instantiate voting contract at address.
+//   const voting = Voting.at(someAddress);
+//
+// findEvent, assertEventEmitted, and assertEventNotEmitted usage:
+//   const { findEvent, assertEventEmitted, assertEventNotEmitted } = hre;
+//   ...
+//   const result = voting.methods.someMethod(arg1, arg2, ...).send({ from: accounts[0] });
+//   const matchOrUndefined = findEvent(result, voting, "SomeEventName",
+//                            eventReturnValues => eventReturnValues.someValueICareAbout === 5);
+//   await assertEventEmitted(result, voting, "SomeEventName",
+//                            eventReturnValues => eventReturnValues.someValueICareAbout === 5);
+//   await assertEventNotEmitted(result, voting, "SomeEventName",
+//                            eventReturnValues => eventReturnValues.someValueICareAbout === 5);
+//
+
+const { extendEnvironment } = require("hardhat/config");
+
+extendEnvironment((hre) => {
+  hre._artifactCache = {};
+  hre.getContract = (name) => {
+    if (!hre._artifactCache[name]) hre._artifactCache[name] = hre.artifacts.readArtifactSync(name);
+    const artifact = hre._artifactCache[name];
+
+    const deployed = async () => {
+      const deployment = await hre.deployments.get(name);
+      return new hre.web3.eth.Contract(artifact.abi, deployment.address);
+    };
+
+    const newProp = (...args) =>
+      new hre.web3.eth.Contract(artifact.abi, undefined).deploy({ data: artifact.bytecode, arguments: args });
+
+    const at = (address) => new hre.web3.eth.Contract(artifact.abi, address);
+
+    return {
+      ...artifact,
+      deployed,
+      new: newProp,
+      at,
+    };
+  };
+
+  hre.findEvent = async (txnResult, contract, eventName, fn = () => true) => {
```

Comment:
What is the expected shape of this `txnResult`? Since we're not building Truffle objects, I'm guessing its not like [this TransactionResult defined in truffle-assertions](https://github.com/rkalis/truffle-assertions#truffleassertcreatetransactionresultcontract-transactionhash)?

Reply:
It is not. It is the web3 transaction result.

```
const txnResult = someContract.methods.someMethod(arg1, arg2, ...).send({ from: accounts[0] });
const event = await findEvent(txnResult, someContract, "MyEvent", someFunction);
```